crates/core/tedge_mapper/src/c8y_converter.rs

```diff
@@ -79,6 +75,51 @@ impl Converter for CumulocityConverter {
         Ok(vec)
     }
 
+    /// [1, 2, 3] = iter1
+    /// [4, 5, 6] = iter2
+    /// iter1.zip(iter2) => [(1, 4), (2, 5), (3, 6))]
+    fn try_convert_alarm(&self, input: &Message) -> Result<Vec<Message>, ConversionError> {
+        let c8y_alarm_topic = Topic::new_unchecked("c8y/s/us");
+        let mut vec: Vec<Message> = Vec::new();
+
+        let topic_name = input.topic.name.as_str();
+        let topic_split: Vec<&str> = topic_name.split('/').collect();
+        if topic_split.len() == 4 {
+            let alarm_name = topic_split[3];
+            let alarm_severity = topic_split[2];
+
+            let c8y_json_payload = smartrest::from_thin_edge_alarm_json(
+                alarm_name,
+                alarm_severity,
+                input.payload_str()?,
+            )?;
```

Comment:
Why not push the topic parsing logic inside the `smartrest::from_thin_edge_alarm_json` method?